x-pack/metricbeat/module/aws/sqs/sqs.go

```diff
@@ -116,14 +114,20 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 		metricDataResults, err := aws.GetMetricDataResults(metricDataQueries, svcCloudwatch, startTime, endTime)
 		if err != nil {
 			err = errors.Wrap(err, "GetMetricDataResults failed, skipping region "+regionName)
-			m.logger.Error(err.Error())
+			m.Logger().Error(err.Error())
 			report.Error(err)
 			continue
 		}
 
 		// Create Cloudwatch Events for SQS
-		createSQSEvents(queueURLs, metricDataResults, regionName, report)
+		err = createSQSEvents(queueURLs, metricDataResults, regionName, report)
+		if err != nil {
+			m.Logger().Debug("Error trying to emit event")
+			return nil
```

Comment:
Shouldn't it be `continue` here so it continues to the other regions. Or is an error returned if publish event is false?

Reply:
The only time the error here is not nil is when report.Event failed, which means the module is being stopped. So instead of `continue` to other regions to collect data which will not be used, `return` seems to be appropriate here.